packages/components/src/components/Select/useSelectData.tsx

```diff
@@ -53,42 +108,60 @@ export const useSelectData = (children: ReactNode) => {
                     }
                 }
             });
-
-            return {
-                inputSlots,
-                menuSlots,
-                clearButton,
-                itemValues: getRecursiveOptionValues(menuSlots),
-            };
         } else {
-            return {
-                menuSlots: children,
-                inputSlots: [],
-                itemValues: getRecursiveOptionValues(children),
-            };
+            menuSlots.push(children);
         }
-    }, [children]);
-
-    const filteredItems = useMemo(
-        () =>
-            itemValues.filter(
-                (item) => filterText === '' || item.label.toLowerCase().includes(filterText.toLowerCase()),
-            ),
-        [itemValues, filterText],
-    );
+
+        if (shouldAddCustomItem) {
+            const customItem = (
+                <ForwardedRefSelectItem key={filterText} value={filterText} data-test-id="custom-item">
+                    {filterText}
+                </ForwardedRefSelectItem>
+            );
+
+            menuSlots.push(customItem);
+        }
+
+        return {
+            inputSlots,
+            menuSlots,
+            clearButton,
+        };
+    }, [children, filterText, shouldAddCustomItem]);
 
     const getItemByValue = useCallback(
         (value?: string) => (value ? itemValues.find((item) => item.value === value) : undefined),
         [itemValues],
     );
 
+    const items = useMemo(() => {
+        const filteredItems = itemValues.filter(
+            (item) => filterText === '' || item.label.toLowerCase().includes(filterText.toLowerCase()),
```

Comment:
could be nice to improve this search such that it splits each words and looks for instances of that word rather than inclusions

Reply:
could be done in the future, for now it's out of scope

Reply:
for sure, just a thought